corehq/apps/es/utils.py

```diff
@@ -193,3 +193,27 @@ def get_es_reindex_setting_value(name, default):
     if ES_REINDEX_LOG[-1] != getattr(settings, 'ES_MULTIPLEX_TO_VERSION', None):
         return default
     return getattr(settings, name, default)
+
+
+def get_user_domain_memberships(user):
+    memberships = []
+    if user['doc_type'] == 'CommCareUser':
+        memberships.append(user['domain_membership'])
+    elif user['doc_type'] == 'WebUser':
+        for membership in user['domain_memberships']:
+            memberships.append(membership)
+    return memberships
+
+
+def populate_user_domain_memberships():
+    from corehq.apps.es.users import UserES, user_adapter
+    user_ids = UserES().get_ids()
+
+    for user in user_adapter.iter_docs(user_ids):
```

Comment:
You might try `scroll_ids_to_disk_and_iter_docs` for this instead - not sure if we'll be able to fetch all user IDs on production in a single request.